lib/shared/bucketing-test-data/src/data/testData.ts

```diff
@@ -49,6 +49,20 @@ export const audiences: PublicAudience[] = [
         _id: '6153557f1ed7bac7268ea0d9',
         filters: {
             filters: [{
+                filters: [{
```

Comment:
any need for a more targeted segmentation test that just unit tests the nested logic?

Reply:
I think it couldnt hurt to add one, so I will

Reply:
it's a lot easier to represent all the different cases the code is supposed to handle as targeted unit tests, rather than burying it implicitly in the test data. Also makes it clearer to see what went wrong when something fails

Reply:
yeah 100p

Reply:
Added a bunch of segmentation level tests